### PR TITLE
Compatibility 231 (#143)

### PR DESCRIPTION
**Pull request by SilverSilence**
# Description
* Update Gradle
* Update Kotlin JVM Version
* Update Intellij Plugin Version
* Update Supported-Until Version
* Update JDK use in Github Action
 

```release-note
Add support for IDEA 2023.1
```

# Special notes for your reviewer

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -45,7 +45,7 @@ idea {
 plugins {
     idea
     kotlin("jvm") version "1.7.21"
-    id("org.jetbrains.intellij") version "1.11.0"
+    id("org.jetbrains.intellij") version "1.13.2"
```
Comment (vgramer) [inline build.gradle.kts:48]:
```suggestion
    id("org.jetbrains.intellij") version "1.13.3"
```
we can bump to the very last version  ;) (released 2 day ago )

File: gradle.properties [hunk 2]
```diff
@@ -9,15 +9,15 @@ kotlin.code.style=official
 baseIDE=idea
 
 # if you change the version of ide, also change psiViewerPluginVersion accordingly (cf https://plugins.jetbrains.com/plugin/227-psiviewer/versions)
-ideaVersion=IC-2022.1
+ideaVersion=IC-2023.1
 pycharmCommunityVersion=PC-2022.1
-psiViewerPluginVersion=221-SNAPSHOT
+psiViewerPluginVersion=231-SNAPSHOT
```
Comment (vgramer) [inline gradle.properties:14]: The plugin should be tested on the oldest version supported
```
ideaVersion=IC-2022.2
pycharmCommunityVersion=PC-2022.2
psiViewerPluginVersion=222-SNAPSHOT
```

File: gradle.properties [hunk 3]
```diff
@@ -9,15 +9,15 @@ kotlin.code.style=official
...
 
 # see https://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html for more information
 sinceBuild=221
```
Comment (vgramer) [inline gradle.properties:17]: We support only the last 3 major releases
```suggestion
sinceBuild=222
```

**Review by vgramer — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 2 above)
Comment (SilverSilence) [reply gradle.properties:14]: It seems that `IC-2022.2` is not compatible with `PsiViewerPluginVersion=231-SNAPSHOT`.

Comment (vgramer) [reply gradle.properties:14]: the versions should be bumped as follow:
```
ideaVersion=IC-2022.2
pycharmCommunityVersion=PC-2022.2
psiViewerPluginVersion=222-SNAPSHOT
```

idea and pycharm are bumped to `2022.2` and psiViewerPluginVersion is bumped to` 222-SNAPSHOT`.

by the way, the PsiViewerPluginVersion is the version of psi viewer plugin to install in the testing ide. So it should match the version of the testing ide.

Comment (SilverSilence) [reply gradle.properties:14]: With your configurations, I get this error:

`Plugin 'PsiViewer:222-SNAPSHOT' is not compatible to: IC-221.5080.210`

Comment (SilverSilence) [reply gradle.properties]: Seems I copied it wrong, my bad.. Works now. I'm sorry that this took several iterations.

Comment (vgramer) [reply gradle.properties]: np :)

**Comment by SilverSilence** [comment]
I don't know how to resolve this error `Must not use `executable` property on `Test` together with `javaLauncher` property` 😔

**Comment by vgramer** [comment]
I have an idea of what it. I need to do some testing. I will let you know

**Comment by vgramer** [comment]
the version of gradle nee

> I don't know how to resolve this error `Must not use "executable" property on "Test" together with "javaLauncher" property` 😔

To fix it, I had to bump the version of Gradle. I also took the opportunity to bump the Kotlin jvm  version and JDK used in the Github action

**Review by vgramer — APPROVED** [review]

**Comment by vgramer** [comment]
@SilverSilence  Thanks for contributing to the project :)

**Pull request merged**